Symfony/CS/Fixer/PSR2/ClassDefinitionFixer.php

```diff
@@ -0,0 +1,218 @@
+
+/**
+ * Fixer for part of the rules defined in PSR2 ¶4.1 Extends and Implements.
+ *
+ * @author SpacePossum
+ */
+final class ClassDefinitionFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+        for ($index = $tokens->getSize() - 1; $index > 0;--$index) {
+            if (!$tokens[$index]->isClassy()) {
+                continue;
+            }
+
+            $this->fixClassDefinition($tokens, $index, $tokens->getNextTokenOfKind($index, array('{')));
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $start  Class definition token start index
+     * @param int    $end    Class definition token end index
+     */
+    private function fixClassDefinition(Tokens $tokens, $start, $end)
+    {
+        // check if there is a `implements` part in the definition, since there are rules for it in PSR 2.
+        $implementsInfo = $this->detectImplements($tokens, $start, $end);
+
+        // 4.1 The extends and implements keywords MUST be declared on the same line as the class name.
+        if ($implementsInfo['numberOfInterfaces'] > 1 && $implementsInfo['multiLine']) {
+            $end += $this->ensureWhiteSpaceSeparation($tokens, $start, $implementsInfo['implementsAt']);
+            $end += $this->fixMultiLineImplements($tokens, $implementsInfo['implementsAt'], $end);
+        } else {
+            $end += $this->ensureWhiteSpaceSeparation($tokens, $start, $end);
+        }
+
+        // 4.1 The opening brace for the class MUST go on its own line;
+        $this->ensureLineBreakBeforeToken($tokens, $end);
+    }
+
+    /**
+     * Fix spacing between lines following `implements`.
+     *
+     * PSR2 4.1 Lists of implements MAY be split across multiple lines, where each subsequent line is indented once.
+     * When doing so, the first item in the list MUST be on the next line, and there MUST be only one interface per line.
+     *
+     * @param Tokens $tokens
+     * @param int    $implementsAt
+     * @param int    $end
+     *
+     * @return int number tokens inserted by the method before the end token
+     */
+    private function fixMultiLineImplements(Tokens $tokens, $implementsAt, $end)
+    {
+        $added = 0;
+        // implements should be followed by a line break, but we allow a comments before that,
+        // the lines after 'implements' are always build up as (comment|whitespace)*T_STRING{1}(comment|whitespace)*','
+        // after fixing it must be (whitespace indent)(comment)*T_STRING{1}(comment)*','
+        for ($c = $end - 1; $c > $implementsAt - 1; --$c) {
+            if ($tokens[$c]->isWhitespace()) {
+                if ($tokens[$c + 1]->equals(',')) {
+                    $tokens[$c]->clear();
+                } elseif ($tokens[$c + 1]->isComment() && !$tokens[$c]->equals(' ')) {
```

Comment:
hmm, does this work?? `->equals(' ')`?
should be always `false`, because `' ' ` is not a valid token prototype.